Source/WebKit/UIProcess/Inspector/win/WebInspectorUIProxyWin.cpp

```diff
@@ -73,6 +74,63 @@ static InspectedWindowInfo getInspectedWindowInfo(HWND inspectedWindow, HWND par
     return { rect.left, rect.top, rect.right - rect.left, rect.bottom - rect.top, parentRect.right - parentRect.left, parentRect.bottom - parentRect.top };
 }
 
+static String systemErrorMessage(DWORD errorCode)
+{
+    if (!errorCode)
+        return emptyString();
+
+    wchar_t* messageBuffer = nullptr;
+    size_t size = FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+        nullptr, errorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_NEUTRAL), reinterpret_cast<LPWSTR>(&messageBuffer), 0, nullptr);
+    String message(messageBuffer, size);
+    LocalFree(messageBuffer);
+
+    return message;
+}
+
+void WebInspectorUIProxy::showSavePanelForSingleFile(HWND parentWindow, Vector<WebCore::InspectorFrontendClient::SaveData>&& saveDatas)
+{
+    // Remove custom URI schemes such as "web-inspector://" and also remove the leading "/".
+    URL url { saveDatas[0].url };
+    auto filePath = url.path().substring(1).toString().wideCharacters();
+    filePath.grow(MAX_PATH);
+
+    OPENFILENAME ofn { };
+    ofn.lStructSize = sizeof(OPENFILENAME);
+    ofn.hwndOwner = parentWindow;
+    ofn.lpstrFile = filePath.data();
+    ofn.nMaxFile = MAX_PATH;
+    ofn.Flags = OFN_PATHMUSTEXIST | OFN_OVERWRITEPROMPT;
+
+    if (GetSaveFileName(&ofn)) {
+        auto fd = FileSystem::openFile(filePath.data(), FileSystem::FileOpenMode::ReadWrite);
+        if (!FileSystem::isHandleValid(fd))
+            return;
+
+        auto content = saveDatas[0].content.utf8();
+        auto contentSize = content.length();
+        auto bytesWritten = FileSystem::writeToFile(fd, content.data(), contentSize);
+        if (bytesWritten == -1 || bytesWritten != contentSize) {
+            auto errMsg = systemErrorMessage(GetLastError());
+            if (errMsg.isEmpty())
+                errMsg = makeString("Error: writeToFile returns ", bytesWritten, ", contentLength = ", content.length());
+            MessageBox(parentWindow, errMsg.wideCharacters().data(), L"Export HAR", MB_OK | MB_ICONEXCLAMATION);
+        }
+        FileSystem::closeFile(fd);
+    } else {
+        String errMsg;
```

Comment:
Move errMsg into the following if block.
It might better to rename this variable. https://webkit.org/code-style-guidelines/#names-full-words
`message` or `text` or something?